ring/batch_test.go

```diff
@@ -69,9 +69,7 @@ func executeScenario(bt batchTracker, scenarios []scenario) error {
 		}
 	}()
 
-	go func() {
-		wg.Wait()
-	}()
+	wg.Wait()
```

Comment:
Why do we need to wait on the `WaitGroup` at all? Won't the `select` below block?

If you want to ensure the goroutine terminates, perhaps moving this below the `select` would be best. Or this could even be simplified to have the goroutine close a channel once the loop is complete, and then the test can wait for that channel to be closed.